internal/mempool/mempool.go

```diff
@@ -654,40 +654,40 @@ func (mp *TxPool) hasMempoolInput(tx *dcrutil.Tx) bool {
 	return false
 }
 
-// fetchRedeemers returns all transactions that reference an outpoint for
-// the provided regular transaction `tx`. Returns nil if a non-regular
-// transaction is provided.
+// fetchRedeemers returns a set of transactions of type redeemerType from
+// redeemerPool that have an input referencing the provided regular transaction.
 //
 // This function MUST be called with the mempool lock held (for reads).
-func (mp *TxPool) fetchRedeemers(outpoints map[wire.OutPoint]*dcrutil.Tx, tx *dcrutil.Tx) []*dcrutil.Tx {
-	txType := stake.DetermineTxType(tx.MsgTx())
-	if txType != stake.TxTypeRegular {
-		return nil
-	}
-
-	tree := wire.TxTreeRegular
+func (mp *TxPool) fetchRedeemers(tx *dcrutil.Tx,
+	redeemerType stake.TxType,
+	redeemerPool map[chainhash.Hash]*TxDesc,
+	redeemerOutpoints map[wire.OutPoint]*dcrutil.Tx) []*TxDesc {
```

Comment:
The function sigature could be formatted to fit on two lines I think. Also since the function states it's fetching redeemers you could shorten the signature by avoiding the `redeemer` prefixes.

Reply:
Removed `redeemer` prefixes from parameter names and reordered the method signature.